lib/adf_info.py

```diff
@@ -614,9 +611,85 @@ def __init__(self, config_file, debug=False):
         #End if
         #Print number of processors being used to debug log (if requested):
         self.debug_log(f"ADF is running with {self.__num_procs} processors.")
-        # -----------------------------------------
 
+        active_env = self.get_active_conda_environment()
```

Comment:
Hi @justin-richling, as discussed in the hackathon, I might recommend the following changes:

1.  It might be good to collect all of the git/conda/config information in an internal ADF object or dictionary as early as possible in the ADF run, maybe even in the `AdfConfig` class?

2.  I might then move all of this code into two functions in `AdfWeb`.  One function can be for when `create_html` is True, which uses this new config info obj/dict to create the new markdown file, and another for when `create_html` is False but the debug log is active, in which case it just writes everything to the logger.

Of course, please let me know if you find that this doesn't actually work, or causes more headaches then benefits.  Thanks!

Reply:
This worked perfectly, thanks!